sql/plan/insert.go

```diff
@@ -432,6 +474,24 @@ func (i *insertIter) ignoreOrClose(ctx *sql.Context, row sql.Row, err error) (sq
 	}
 }
 
+// convertDataAndIgnore modifies a row with data conversion issues in INSERT IGNORE calls
+// Per MySQL docs "Rows set to values that would cause data conversion errors are set to the closest valid values instead"
+// cc. https://dev.mysql.com/doc/refman/8.0/en/sql-mode.html#sql-mode-strict
+func (i *insertIter) convertDataAndIgnore(ctx *sql.Context, row sql.Row, columnIdx int, err error) (sql.Row, error) {
+	if sql.ErrLengthBeyondLimit.Is(err) {
+		maxLength := i.schema[columnIdx].Type.(sql.StringType).MaxCharacterLength()
+		row[columnIdx] = row[columnIdx].(string)[:maxLength] // truncate string
+	} else if sql.ErrInvalidValue.Is(err) {
+		row[columnIdx] = i.schema[columnIdx].Type.Zero()
+	} else {
+		return i.ignoreOrClose(ctx, row, err)
```

Comment:
Don't use this here: the semantics are to return a nil row in some cases, which you definitely do not want from this function. Just return an error and let the caller figure out what to do